src/main/java/com/demo/MybatisApplication/controller/StudentController.java

```diff
@@ -1,49 +1,46 @@
 package com.demo.MybatisApplication.controller;
 
 
-import com.demo.MybatisApplication.dto.StudentAddDto;
-import com.demo.MybatisApplication.dto.StudentDisplayAsSubjects;
-import com.demo.MybatisApplication.dto.StudentDisplayByIdDto;
-import com.demo.MybatisApplication.dto.StudentsDisplayDto;
-import com.demo.MybatisApplication.model.SubjectEntity;
+import com.demo.MybatisApplication.dto.StudentAdditionDto;
+import com.demo.MybatisApplication.dto.StudentSubjectsDisplayDto;
+import com.demo.MybatisApplication.dto.StudentDisplayDto;
 import com.demo.MybatisApplication.service.StudentService;
+import jakarta.validation.Valid;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
-
 import java.util.List;
 
-
 @RestController
 @RequestMapping("/students")
 public class StudentController {
 
-    @Autowired
-    private StudentService studentService;
+   @Autowired
+   private StudentService studentService;
 
     @GetMapping("/{id}")
-
-    public StudentDisplayByIdDto getStudentById(@PathVariable Long id){
+    public ResponseEntity<?> getStudentById(@PathVariable Long id){
         return studentService.getStudentById(id);
     }
 
     @PostMapping("/{studentId}/subjects")
-    public void assignSubjectsToStudent(@PathVariable Long studentId, @RequestBody List<SubjectEntity> subjects) {
-        studentService.assignSubjectsToStudent(studentId, subjects);
+    public ResponseEntity<String> assignSubjectsToStudent(@PathVariable Long studentId,@Valid @RequestBody List<Long> subjectIds) {
```

Comment:
Don't return String, please use DTO as the return type.

Reply:
Done